packages/react/src/components/Tile/Tile.tsx

```diff
@@ -497,48 +502,41 @@ export const SelectableTile = React.forwardRef<
     className
   );
 
-  function handleClick(evt) {
+  // Single function to handle selection changes
+  const handleSelectionChange = useCallback(
+    (evt: React.SyntheticEvent, newSelected: boolean) => {
+      setIsSelected(newSelected);
+      onChange(evt as React.ChangeEvent<HTMLDivElement>, newSelected, id);
+    },
+    [onChange, id]
+  );
+
+  function handleClick(evt: React.MouseEvent<HTMLDivElement>) {
```

Comment:
`MouseEvent` is already imported.

```suggestion
  function handleClick(evt: MouseEvent<HTMLDivElement>) {
```